### ISSUE TITLE
Cannot remove a listed external site entry

### ISSUE THREAD

**Issue by diplodata** [nextcloud/external#112]
I cannot remove an external site I have created.  In the External sites settings I click `Remove site` and nothing happens.  I click it again and this time the entry disappears.  But when I click `External sites` settings icon again, the entry returns.

I have tried removing and reinstalling the app, but the old site is still there.  Grateful for assistance.

**Operating system**: Ubuntu 16.04

**Web server:** Apache2.4

**Database:** MariaDB

**PHP version:** 7.2

**Nextcloud version:**  13.0.5

**List of activated apps:**

Talk, Circles, News, External Sites Pico-CMS, GPXPod, GPXEdit, Music

**Comment by nickvergessen** [comment]
Works fine here.

Anything in your browsers console? (F12)

**Comment by diplodata** [comment]
Thanks Nick. The following line appears in console:

    XHRDELETE    https://sosc.io/ocs/v2.php/apps/external/api/v1/sites/1    [HTTP/1.1 405 Method Not Allowed 84ms]

I expanded the entry and copied the header text:

**Response headers**

```
XHRDELETEhttps://mydomain.org/ocs/v2.php/apps/external/api/v1/sites/1[HTTP/1.1 405 Method Not Allowed 84ms] 
HeadersCookiesParamsResponseTimingsStack TraceRequest URL:Request method:Remote address:Status code:405Edit and ResendRaw headersVersion:Response headers (836 B)
Cache-Controlno-store, no-cache, must-revalidate
Connectionkeep-aliveContent-Length0
Content-Security-Policyt-src 'none'; base-uri 'self';
Content-Typetext/html; charset=UTF-8
DateThu, 02 Aug 2018 21:42:27 GMT
ExpiresThu, 19 Nov 1981 08:52:00 GMT
Pragmano-cacheServernginx/1.12.1
X-Content-Type-Optionsnosniff
X-Download-Optionsnoopen
X-Frame-OptionsSAMEORIGIN
X-Permitted-Cross-Domain-Policiesnone
X-Robots-Tagnone
X-XSS-Protection1; 
mode=blockRequest headers (770 B)Acceptapplication/json, text/javascript, */*; q=0.01Accept-Encodinggzip, deflate, brAccept-Languageen-US,en;q=0.5Connectionkeep-aliveCookieoc8bon5u9snl=k1d188uo4grujco83…-nc_sameSiteCookiestrict=trueDNT1Hostmydomain.orgOCS-APIREQUESTtruerequesttokenlFjiQXMxaUiZD3XQJB3HA411fTeJnE…AQC+Md98PGl6+qQ5Lxcw6QE93Bmo=User-AgentMozilla/5.0 (Macintosh; Intel …) Gecko/20100101 Firefox/62.0X-Requested-WithXMLHttpRequest
```

**Request headers**

```
XHRDELETEhttps://mydomain.org/ocs/v2.php/apps/external/api/v1/sites/1[HTTP/1.1 405 Method Not Allowed 84ms] 
HeadersCookiesParamsResponseTimingsStack TraceRequest URL:Request method:Remote address:Status code:405Edit and ResendRaw headersVersion:Response headers (836 B)Cache-Controlno-store, no-cache, must-revalidateConnectionkeep-aliveContent-Length0Content-Security-Policyt-src 'none'; base-uri 'self';Content-Typetext/html; charset=UTF-8DateThu, 02 Aug 2018 21:42:27 GMTExpiresThu, 19 Nov 1981 08:52:00 GMTPragmano-cacheServernginx/1.12.1X-Content-Type-OptionsnosniffX-Download-OptionsnoopenX-Frame-OptionsSAMEORIGINX-Permitted-Cross-Domain-PoliciesnoneX-Robots-TagnoneX-XSS-Protection1; mode=blockRequest headers (770 B)
Acceptapplication/json, text/javascript, */*; q=0.01
Accept-Encodinggzip, deflate, br
Accept-Languageen-US,en;q=0.5
Connectionkeep-alive
Cookieoc8bon5u9snl=k1d188uo4grujco83…-nc_sameSiteCookiestrict=true
DNT1
Hostmydomain.org
OCS-APIREQUESTtrue
requesttokenlFjiQXMxaUiZD3XQJB3HA411fTeJnE…AQC+Md98PGl6+qQ5Lxcw6QE93Bmo=
User-AgentMozilla/5.0 (Macintosh; Intel …) Gecko/20100101 Firefox/62.0
X-Requested-WithXMLHttpRequest
```

**Comment by GlasierXplor** [comment]
Hi I am also suffering from the same issue
## What I have tried
1. Remove and reinstalled External Sites
2. Spamming the delete buttons 

## What actually happened
1. The sites keep coming back (as the bug has reported).
2. The sites duplicated themselves (now there are 4).

## What workarounds can I think of
1. Manually remove the extra sites

## Why can't I do it
1. I cannot seem to find where the configuration files that handle the external sites are located.

## What anyone can help me with
1. Help to see if the bug can be fixed.
2. Teach us how to remove the extra sites manually.

**Comment by dgferrin** [comment]
This is the config in external sites using occ command with no changes in the app yet:

"external": {
            "enabled": "yes",
            "installed_version": "3.2.0",
            "max_site": "7",
            "types": ""

And this is the config with for example 1 link added:

"external": {
            "enabled": "yes",
            "installed_version": "3.2.0",
            "max_site": "8",
            "sites": "{\"8\":{\"id\":8,\"name\":\"mio\",\"url\":\"http:\\\/\\\/google.es\",\"lang\":\"\",\"type\":\"link\",\"device\":\"\",\"icon\":\"external.svg\",\"groups\":[],\"redirect\":false}}",
            "types": ""

Im having the same issue, so i can't remove sites as expected.

So what i did is remove it manually by using this command:
occ config:app:delete external sites

Dunno if it has any repercussion or not but at least i was able to delete links.

**Comment by iSpeeX** [comment]
You are right the "occ config:app:delete external sites" command is enough to remove the problem... but just for a moment.
At first, everything worked well again, but after 2 sites added, same problem. Now I can't edit any setting, it stays in "grey" state. There's definitely a problem.

**Comment by archive-r** [comment]
same issue

**Comment by joshtrichards** [comment]
> XHRDELETE    https://sosc.io/ocs/v2.php/apps/external/api/v1/sites/1    [HTTP/1.1 405 Method Not Allowed 84ms

At least in OP's case, this looks like a configuration matter.

`DELETE` should be permitted. It's being blocked by web server configuration or a WAF (like mod_security) or something along those lines probably.

Everyone else, check browser inspect (re-attempting while the *Console* tab is open and also again while the *Network* tab is open). Also suggest doing with a full refresh (e.g. <shift>-<refresh>).
